ansible_collections/arista/avd/roles/eos_designs/python_modules/underlay/utils.py

```diff
@@ -184,3 +188,108 @@ def _get_l3_interface_cfg(self, l3_interface: dict) -> dict | None:
+            "ip_address": svi.get("ip_address"),
+            "ipv6_address": svi.get("ipv6_address"),
+            "ipv6_enable": svi.get("ipv6_enable"),
+            "mtu": svi.get("mtu") if self.shared_utils.platform_settings_feature_support_per_interface_mtu else None,
+            "eos_cli": svi.get("raw_eos_cli"),
+            "struct_cfg": svi.get("structured_config"),
+        }
+        if (mtu := subinterface["mtu"]) is not None and subinterface["mtu"] > self.shared_utils.p2p_uplinks_mtu:
+            raise AristaAvdError(
+                f"MTU '{self.shared_utils.p2p_uplinks_mtu}' set for 'p2p_uplinks_mtu' must be larger or equal to MTU '{mtu}' "
+                f"set on the SVI '{svi_id}'."
+                "Either adjust the MTU on the SVI or p2p_uplinks_mtu."
+            )
+
+        # Only set VRRPv4 if ip_address is set
+        if subinterface["ip_address"] is not None:
+            # TODO in separate PR adding VRRP support for SVIs
+            pass
+
+        # Only set VRRPv6 if ipv6_address is set
+        if subinterface["ipv6_address"] is not None:
+            # TODO in separate PR adding VRRP support for SVIs
+            pass
+
+        svi_ip_helpers: list[dict] = convert_dicts(default(svi.get("ip_helpers"), vrf.get("ip_helpers"), []), "ip_helper")
+        if svi_ip_helpers:
+            subinterface["ip_helpers"] = [
+                {"ip_helper": svi_ip_helper["ip_helper"], "source_interface": svi_ip_helper.get("source_interface"), "vrf": svi_ip_helper.get("source_vrf")}
+                for svi_ip_helper in svi_ip_helpers
+            ]
+
+        if get(svi, "ospf.enabled") is True and get(vrf, "ospf.enabled") is True:
+            subinterface.update(
+                {
+                    "ospf_area": svi["ospf"].get("area", "0"),
+                    "ospf_network_point_to_point": svi["ospf"].get("point_to_point", False),
+                    "ospf_cost": svi["ospf"].get("cost"),
+                }
+            )
+            ospf_authentication = svi["ospf"].get("authentication")
+            if ospf_authentication == "simple" and (ospf_simple_auth_key := svi["ospf"].get("simple_auth_key")) is not None:
+                subinterface.update({"ospf_authentication": ospf_authentication, "ospf_authentication_key": ospf_simple_auth_key})
+            elif ospf_authentication == "message-digest" and (ospf_message_digest_keys := svi["ospf"].get("message_digest_keys")) is not None:
+                ospf_keys = []
+                for ospf_key in ospf_message_digest_keys:
+                    if not ("id" in ospf_key and "key" in ospf_key):
+                        continue
+
+                    ospf_keys.append({"id": ospf_key["id"], "hash_algorithm": ospf_key.get("hash_algorithm", "sha512"), "key": ospf_key["key"]})
+                if ospf_keys:
+                    subinterface.update({"ospf_authentication": ospf_authentication, "ospf_message_digest_keys": ospf_keys})
```

Comment:
is this something we have already in network services for SVIs that we could unify?

Reply:
This is only partially overlapping with SVIs. Some elements have been tweaked. Of course we could fish out the common parts and put in shared_utils if we feel it is worth it.

Reply:
done. Please review again.